sound/soc/sof/intel/cnl.c

```diff
@@ -40,16 +39,14 @@ static irqreturn_t cnl_ipc_irq_thread(int irq, void *context)
 	irqreturn_t ret = IRQ_NONE;
 
 	hipcida = snd_sof_dsp_read(sdev, HDA_DSP_BAR, CNL_DSP_REG_HIPCIDA);
-	hipcctl = snd_sof_dsp_read(sdev, HDA_DSP_BAR, CNL_DSP_REG_HIPCCTL);
 	hipctdr = snd_sof_dsp_read(sdev, HDA_DSP_BAR, CNL_DSP_REG_HIPCTDR);
 
 	/* reenable IPC interrupt */
 	snd_sof_dsp_update_bits(sdev, HDA_DSP_BAR, HDA_DSP_REG_ADSPIC,
 				HDA_DSP_ADSPIC_IPC, HDA_DSP_ADSPIC_IPC);
 
 	/* reply message from DSP */
-	if (hipcida & CNL_DSP_REG_HIPCIDA_DONE &&
-	    hipcctl & CNL_DSP_REG_HIPCCTL_DONE) {
+	if (hipcida & CNL_DSP_REG_HIPCIDA_DONE) {
```

Comment:
NACK on this, we added this to fix bugs that treated new messages as reply messages by mistake, your change is going backwards.

Reply:
NAK on your comment then. You did not provide any justification for the existing code, and I will assert that it's not possible to have an interrupt WITHOUT those HIPPCCTL_BUSY/DONE set. You must be confusing source and effect.

Reply:
@keyonjie I have always found your patch to add this change odd. HIPPCCTL_BUSY/DONE is set only to allow the  DONE interrupt from the DSP to be propagated to the host. @plbossart is right here.

Reply:
@plbossart @ranj063 we added the changes to fix real issue, please refer to commit b50619398c7befa:
```
    ASoC: SOF: hda-ipc: fix IPC handled wrong issue
    
    When IPC interrupt arrived, we need to check HDA_DSP_REG_HIPCCTL to make
    sure we can tell the message initiator(new or reply from DSP),
    otherwise, we may treat new message as reply one by mistake, and see
    error message reported as "error: rx list empty but received ...".
```

To me, before Pierre's this PR, we still need that **odd** fix, you can imagine scenario happens like this:
1. FW replied IPC to host, and handling the "reply message" section in the irq_thread, be noticed that IPC interrupt is already re-enabled here.
2. FW initiated IPC to host, we should handle it in the "new message" section in the irq_thread, but as HDA_DSP_REG_HIPCIE_DONE is not cleared yet, without the **odd** fix, it will go into the "reply message" section by mistake!

But after think it more, this **odd** thing can actually be removed if we apply the commit "re-enable IPC IRQ at end of handler", so moving this "ipc: don't check for HIPCCTL register value" commit after the "re-enable IPC IRQ at end of handler" commit looks good to me.

Reply:
@keyonjie are you sure we still have this issue? Can we possibly revert your commit and check?

Reply:
@keyonjie the commit b506193 looks like a mistake to me. There is absolutely no way you can use the interrupt mask to know if this is a reply or not. Wrong register, wrong conclusion. You can only check if you enabled the BUSY and DONE interrupts.

And if you use it to trap an error case due to a race condition, then we are doing something absolutely out of control, which generates a risk of either losing an interrupt or receiving fake ones. Either way it's not not so good.

If that makes you more comfortable I can more the first patch to last place in the series.

Reply:
> @keyonjie the commit [b506193](https://github.com/thesofproject/linux/commit/b50619398c7befa3bc19266c352d0bf4cc51d948) looks like a mistake to me. There is absolutely no way you can use the interrupt mask to know if this is a reply or not. Wrong register, wrong conclusion. You can only check if you enabled the BUSY and DONE interrupts.

So here "the DONE interrupts" in "You can only check if you enabled the BUSY and DONE interrupts" is actually the interrupt mask HDA_DSP_REG_HIPCCTL_DONE?

> 
> And if you use it to trap an error case due to a race condition, then we are doing something absolutely out of control, which generates a risk of either losing an interrupt or receiving fake ones. Either way it's not not so good.
> 
> If that makes you more comfortable I can more the first patch to last place in the series.

Reply:
yes. HIPCCTL controls the propagation of interrupts to the host only.